src/api/_queries/story.query.ts

```diff
@@ -23,3 +25,34 @@ export const storyCommonFields: string = `
     approximateReadingTime,
     mediaSources
 `;
+
+export const storiesByAuthorSlugQuery = groq`*[_type == 'story' && author->slug.current == $slug][$slice]
```

Comment:
¿Tenemos forma de "estructurar" esta query usando los consts que están utilizando algunas otras queries ya existentes? Especialmente creo que sería oportuno poder evitarnos el escribir las subconsultas de `mediaSources` y `resources` aquí de nuevo.

Reply:
AFAIK y lo que busqué... nop

Armé un post en SO también pero no tuve respuestas https://stackoverflow.com/questions/78639610/reusing-pieces-of-sanity-groq-queries-and-still-being-able-to-use-the-typegen-to

El problema es que si interpolás el valor de la query no lo detecta el typegen de sanity para groq queries. Probé de ambas formas y si usaba variables dentro no las detectaba :/

Y siguiendo lo que aclara en las docs tiene sentido:
![image](https://github.com/cuentoneta/cuentoneta/assets/35072686/4ebff2e4-de73-4f9e-b2f0-fca7c4524e00)
https://www.sanity.io/learn/course/typescripted-content/generating-type-for-groq-query-results

Reply:
A juzgar por el punto 3 vamos a tener que usar consultas 100% en literals. Creo que el tradeoff entre tener las queries tipadas y tener que expresarlas en literales vale la pena, así que procedamos con escribirlas de esta manera a partir de ahora.